cloudant-client/src/main/java/com/cloudant/client/org/lightcouch/CouchDbException.java

```diff
@@ -81,10 +76,11 @@ public String getReason() {
     @Override
     public String getMessage() {
         //improve the message with status code, error and reason if we have them
-        return ((getStatusCode() > 0) ? getStatusCode() + " " : "") + super.getMessage()
-                + ((url != null) ? ": " + url : "")
-                + ((error != null) ? ": " + getError() : "")
-                + ((reason != null) ? ": " + getReason() : "");
+        String msg = super.getMessage().replaceAll("\\.$", "");
```

Comment:
Whilst this is tidy, I wonder if it might be a bit inefficient to use a regex.
In my simplistic testing using:
`msg = (super.getMessage().endsWith(".")) ? super.getMessage().substring(0, msg.length() - 1) : super.getMessage();`

is around 13 times faster when the replacement needs to be made, and more than 20 times faster in the case when it doesn't. Whilst hopefully we won't be writing exception messages often it might be worth considering.